src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1720,6 +1752,94 @@ private void AppendSymbolsWithName(ArrayBuilder<ISymbol> results, string name, B
             }
         }
 
+#nullable enable
+        private void AppendExtensionSymbolsWithName(ArrayBuilder<ISymbol> results, string name, Binder binder, TypeSymbol container, TypeSymbol? skipType, LookupOptions options, LookupSymbolsInfo info)
+        {
+            Debug.Assert(skipType is null || (skipType.IsExtension && skipType.IsDefinition));
+
+            LookupSymbolsInfo.IArityEnumerable? arities;
+            Symbol? uniqueSymbol;
+
+            if (info.TryGetAritiesAndUniqueSymbol(name, out arities, out uniqueSymbol))
+            {
+                if (uniqueSymbol is not null)
+                {
+                    // This name mapped to something unique.  We don't need to proceed
+                    // with a costly lookup.  Just add it straight to the results.
+                    addSymbolUnlessSkipped(results, uniqueSymbol, skipType);
+                }
+                else
+                {
+                    if (arities is null)
+                    {
+                        throw ExceptionUtilities.Unreachable();
+                    }
+
+                    // The name maps to multiple symbols. Actually do a real lookup so 
+                    // that we will properly figure out hiding and whatnot.
+                    foreach (var arity in arities)
+                    {
+                        appendExtensionSymbolsWithNameAndArity(results, name, arity, binder, container, skipType, options, this.Root);
+                    }
+                }
+            }
+
+            return;
+
+            static void appendExtensionSymbolsWithNameAndArity(ArrayBuilder<ISymbol> results, string name, int arity, Binder binder, TypeSymbol container, TypeSymbol? skipType, LookupOptions options, CSharpSyntaxNode root)
+            {
+                Debug.Assert(results != null);
+
+                // Don't need to de-dup since AllMethodsOnArityZero and AllNamedTypesOnArityZero can't be set at this point.
+                Debug.Assert((options & LookupOptions.AllMethodsOnArityZero) == 0);
+                Debug.Assert((options & LookupOptions.AllNamedTypesOnArityZero) == 0);
+
+                var lookupResult = LookupResult.GetInstance();
+                var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+                binder.LookupMembersInExtensions(lookupResult, name, arity, basesBeingResolved: null, ref discardedUseSiteInfo, options, container);
+
+                if (lookupResult.IsMultiViable)
+                {
+                    if (lookupResult.Symbols.Any(t => t.Kind == SymbolKind.NamedType || t.Kind == SymbolKind.Namespace || t.Kind == SymbolKind.ErrorType))
+                    {
+                        // binder.ResultSymbol is defined only for type/namespace lookups
```

Comment:
Do error types need to go through this path?

Reply:
Added assertion. Neither error types or namespaces should reach here